### PR TITLE
overlay2: remove deprecated overlay2.override_kernel_check option

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to / follow-up to https://github.com/moby/moby/pull/44278
- relates to / follow-up to https://github.com/moby/moby/pull/44279

This option was deprecated in e35700eb5018ff14d91bf5e46f39e65cfe711b6f (and backported to v23.0 through 43ce8f7d24dc949949f0332e30a57dde3a1d6f82).

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: daemon/graphdriver/overlay2/overlay.go [hunk 1]
```diff
@@ -234,9 +234,6 @@ func parseOptions(options []string) (*overlayOptions, error) {
 		}
 		key = strings.ToLower(key)
 		switch key {
-		case "overlay2.override_kernel_check":
-			// TODO(thaJeztah): change this to an error, see https://github.com/docker/cli/pull/3806
```
Comment (tianon) [inline daemon/graphdriver/overlay2/overlay.go:238]: Should we change this to an error, as your comment suggests? :sweat_smile:

Comment (thaJeztah) [reply daemon/graphdriver/overlay2/overlay.go:238]: Yeah, I was on the fence; do we need a specific error, or is the `fmt.Errorf("overlay2: unknown option %s", key)` good enough?

I had some variation of the deprecation message at first (but then we would need another TODO to remove it (?) 😂 

Happy to hear what you (and others) think.

Comment (tianon) [reply daemon/graphdriver/overlay2/overlay.go:238]: Oh, we already have a fallback error; sorry I missed that!

**Review by tianon — APPROVED** [review]

File: daemon/graphdriver/overlay2/overlay.go (hunk 1 above)
Comment (thaJeztah) [reply daemon/graphdriver/overlay2/overlay.go:238]: Yes, it's definitely less "nice" than a dedicated error, but considering the options, we were already printing a warning (so it _shouldn't_ come as a surprise), and I _doubt_ there's enough users of this option to warrant adding a special case, so KISS 😅

**Review by vvoland — APPROVED** [review]

**Pull request merged**